class3.2/include/test.cpp

```diff
@@ -0,0 +1,76 @@
+#include "UnitTests.hpp"
+#include <cmath>
+#include <gtest/gtest.h> 
+
+class Triangle {
+public:
+    Triangle(double a, double b, double c) : side1(a), side2(b), side3(c) {}
+
+    double area() const {
+        double s = (side1 + side2 + side3) / 2.0;
+        return std::sqrt(s * (s - side1) * (s - side2) * (s - side3));
+    }
+
+private:
+    double side1, side2, side3;
+};
+
+void test_triangle_area() {
+    // Test cases for triangle area
+    Triangle triangle1(3, 4, 5);
+    ASSERT_EQ(triangle1.area(), 6.0);
+
+    Triangle triangle2(7, 8, 9);
+    ASSERT_EQ(triangle2.area(), 26.832815729997478);
+
+    Triangle triangle3(10, 10, 10);
+    ASSERT_EQ(triangle3.area(), 43.3012701892);
+}
+
+class Color {
+public:
+    Color(int r, int g, int b) : red(r), green(g), blue(b) {}
+
+    void generateMagicColor() {
+        red = red / 2 - 1;
```

Comment:
what about input validation for r,g,b and the same for modified coordinates. if red is 0 then updated red is -1 which is not ok. The same for green - it can be bigger then 255 after multiplication and it may be -2 if original green is 0. You can use std::clamp to simplify checks and make adjustment ot you can return an error. And constructor should also verify the input somehow